packages/form-builder/src/components/onboarding/Onboarding.tsx

```diff
@@ -0,0 +1,79 @@
+import {useContext, useEffect} from "react";
+import {useDispatch} from "@wordpress/data";
+import {ShepherdTour, ShepherdTourContext} from "react-shepherd";
+import options from './options'
+import steps from './steps'
+
+import "shepherd.js/dist/css/shepherd.css";
+
+declare global {
+    interface Window {
+        onboardingTourData?: {
+            actionUrl: string;
+            autoStartTour: boolean;
+        };
+    }
+}
+
+function TourEffectsAndEvents() {
+    // @ts-ignore
+    const tour = window.tour = useContext(ShepherdTourContext);
+
+    const {selectBlock} = useDispatch('core/block-editor');
+
+    useEffect(() => {
+        const selectAmountBlockCallback = () => {
+            const amountBlock = document.querySelector('[data-type="custom-block-editor/donation-amount-levels"]');
+            const amountBlockId = amountBlock.getAttribute('data-block');
+            selectBlock(amountBlockId).then(() => console.log('Amount block selected'));
+        }
+
+        document.addEventListener('selectAmountBlock', selectAmountBlockCallback );
+
+        return () => {
+            window.removeEventListener('selectAmountBlock', selectAmountBlockCallback);
+        }
+    })
```

Comment:
@kjohnson The way this is written, the effect is going to fire every single time the component renders. Is that the intended behavior?

Reply:
Am I right in guessing you only want this to fire the first time the component mounts?

Reply:
In practice, this doesn't seem to be an issue. This component has no props and doesn't have a reason to re-render.

Reply:
Resolved, see https://github.com/impress-org/givewp-next-gen/pull/143/commits/f54b511974ccf1f77d599ffb746751cf3d6624e7.